File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-202259-53404/output/project/src/main/java/customerimporter/OldZipCode.java
package customerimporter;

import static customerimporter.Validation.assertThat;
import static customerimporter.Validation.isAllDigits;
import static customerimporter.Validation.isBetween;

import java.util.function.Supplier;

public class OldZipCode extends ZipCode {

    private static final int MINIMUM_VALUE = 1000;
    private static final int MAXIMUM_VALUE = 9999;

    private final int value;

    // instance creation

    public static boolean canBeCreatedFrom(String aZipCode) {
        return Character.isDigit(aZipCode.charAt(0));
    }

    public static OldZipCode with(String aZipCode) {
        assertIsValid(aZipCode);
        return new OldZipCode(Integer.parseInt(aZipCode));
    }

    // error messages

    public static String invalidZipCodeErrorDescription() {
        return "Invalid old zipcode";
    }

    // asserting - private

    private static void assertIsValid(String aZipCode) {
        assertThat(isAllDigits(aZipCode), invalidZipCodeErrorDescription());
        assertThat(isBetween(Integer.parseInt(aZipCode), MINIMUM_VALUE, MAXIMUM_VALUE), invalidZipCodeErrorDescription());
    }

    // initialization

    private OldZipCode(int aValue) {
        value = aValue;
    }

    // accessing

    public Object value() {
        return value;
    }

    // testing

    public boolean isNew() {
        return false;
    }

    public boolean isOld() {
        return true;
    }

    // evaluating

    public Object newZipCodeIfNone(Supplier<Object> aNoneBlock) {
        return aNoneBlock.get();
    }

    public Object oldZipCodeIfNone(Supplier<Object> aNoneBlock) {
        return value();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-203929-73794/analysis/own-tests-project/src/customerimporter/old_zip_code.py
from customerimporter.zip_code import ZipCode


class OldZipCode(ZipCode):

    # type

    @classmethod
    def recognizes(cls, an_imported_zip_code):
        return an_imported_zip_code[:1].isdigit()

    def is_old(self):
        return True

    # initialization

    @classmethod
    def _value_from(cls, an_imported_zip_code):
        return int(an_imported_zip_code)

    # zip code

    def old_zip_code_if_none(self, a_none_block):
        return self.value()

    # error messages

    @classmethod
    def invalid_value_error_description(cls):
        return "Invalid old zipcode"

    # validation

    @classmethod
    def assert_is_valid(cls, an_imported_zip_code):
        cls._assert_that(cls._is_old_zip_code_number(an_imported_zip_code))
